contacts/main.js

```diff
@@ -0,0 +1,563 @@
+    if (contactsList[i].id === personId) {
+      contactsList.splice(i, 1);
+      break;
+    }
+}
+
+function incrementLikes(personId) {
+  for (let i = 0; i < contactsList.length; i++)
+    if (contactsList[i].id === personId) {
+      if (contactsList[i].likes < 99) {
+        contactsList[i].likes += 1;
+        break;
+      }
+    }
+}
+
+function decrementLikes(personId) {
+  for (let i = 0; i < contactsList.length; i++)
+    if (contactsList[i].id === personId) {
+      if (contactsList[i].likes > 0) {
+        contactsList[i].likes -= 1;
+        break;
+      }
+    }
+}
+
+///////////////////////validation functions///////////////////////
+
+function validateName(name) {
+  var regex = /^[A-Za-z0-9 ]+$/;
+  var isValid = regex.test(name);
+  if (!isValid)
+    throw new EvalError('Your name contains special characters');
+}
+
+function validateImageUrl(imageUrl) {
+  let pattern = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/;
+  let isValid = pattern.test(imageUrl);
+  if (!isValid)
+    throw new EvalError('Your image is not an url');
+}
+
+/////////////////////////////////////////////////////////////////
+
+function getInput(id) {
+  return document.getElementById(id).value;
+}
+
+function addNewPerson() {
+  try {
+    let newPerson = getTheNewPerson();
+    contactsList.unshift(newPerson);
+    rerenderContacts(contactsList);
+    clearFormInputs();
+    changeAddButtonState();
+    addedSuccessfullyPerson(newPerson);
+  } catch (err) {
+    alert(err.message);
+    clearFormInputs();
+  }
+}
+
+function addedSuccessfullyPerson(newPerson) {
+  alert(newPerson.firstName + ' ' + newPerson.lastName + ' was added successfully');
+}
+
+function clearFormInputs() {
+  document.getElementById('first-name').value = '';
+  document.getElementById('last-name').value = '';
+  document.getElementById('profile-image').value = '';
+}
+
+function getTheNewPerson() {
+  const id = generateUniqueID();
+  const firstName = getInput('first-name');
+  validateName(firstName);
+  const lastName = getInput('last-name');
+  validateName(lastName);
+  const image = getInput('profile-image');
+  validateImageUrl(image);
+  let likes = 0;
+  let obj = {
+    id,
+    firstName,
+    lastName,
+    image,
+    likes
+  };
+  return new Person(obj);
+}
+
+//////////////////////////////////////////////////////////////////////////
+function clearButtonAction() {
+  clearContactsLikes();
+  rerenderContacts();
+  clearFavsListHTML();
+}
+
+function clearContactsLikes() {
+  contactsList.forEach((item) => {
+    item.likes = 0;
+  })
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+function changeAddButtonState() {
+  let first_name = document.getElementById('first-name').value;
+  let last_name = document.getElementById('last-name').value;
+  let image = document.getElementById('profile-image').value;
+  if (first_name != '' && last_name != '' && image != '')
```

Comment:
use !== instead of !=. see https://stackoverflow.com/questions/8616483/javascript-comparison-operators-vs for more details